docs/usage/user.asciidoc

```diff
@@ -0,0 +1,396 @@
+
+Whenever a user is performing an action that affects another user -- typically
+in IAM scenarios -- the user affected by the action is captured at
+`user.target`. The user performing the IAM activity is captured at the root
+of the event.
+
+Examples of IAM activity include:
+
+* user-a creates or deletes user-b
+* user-a modifies user-b
+
+In the create/delete scenarios, there's either no prior state (user creation)
+or no post state (user deletion). In these cases, only `user` at the root and
+`user.target` must be populated.
+
+Example where "root" creates user "bob":
+
+[source,json]
+-----------
+{
+  "user": {
+    "name": "root",
+    "id": "1",
+    "target": {
+      "name": "bob",
+      "id": "1002",
+      ...
+    }
+  }
+  "related": { "user": ["bob", "root"] }
+}
+-----------
+
+When there's a change of state to an existing user, `user.target` must be used
+to capture the prior state of the user, and `user.changes` should list only
+the changes that were performed.
+
+Example where "root" renames user "bob" to "bob.barker":
+
+[source,json]
+-----------
+{
+  "user": {
+    "name": "root",
+    "id": "1",
+    "target": {
+      "name": "bob",
+      "id": "1002"
+    },
+    "changes": {
+      "name": "bob.barker"
+    }
+  },
+  "related": { "user": ["bob", "bob.barker", "root"] }
+}
+-----------
+
+You'll note in the example above that unmodified attributes like the user ID are
+not repeated under `user.changes.*`, since they didn't change.
+
+[discrete]
+[[ecs-user-usage-combining]]
+====== Combining IAM and Privilege Change
+
+We've covered above how `user.target` and `user.changes` can be used at the same time.
+If privilege escalation is also present in the same IAM event, `user.effective`
+should of course be used as well.
+
+Here's the "rename" example from the IAM section above. In the following example,
+we know "alice" is escalating privileges as "root", in order to modify user "bob":
+
+[source,json]
+-----------
+{
+  "user": {
+    "name": "alice",
+    "id": "1001",
+    "effective": {
+      "name": "root",
+      "id": "1"
+    },
+    "target": {
+      "name": "bob",
+      "id": "1002"
+    },
+    "changes": {
+      "name": "bob.barker"
+    }
+  },
+  "related": { "user": ["alice", "bob", "bob.barker", "root"] }
+}
+-----------
+
+[discrete]
+[[ecs-user-usage-reuse-subtleties]]
+====== Subtleties around field reuse
+
+Most cases of field reuse in ECS are reusing a field set inside another field set.
+Two examples of this are:
+
+* reusing `group` in `user`, resulting in the `user.group.*` fields, or
+* reusing `user` in `destination`, resulting in the `destination.user.*` fields,
+  which also include `destination.user.group.*`.
+
+The `user` fields can also be reused within `user` as different names,
+representing the role of each relevant user. Examples are the `user.target.*` or `user.effective.*` fields.
+
+It's important to note, that contrary to the `group` fields,
```

Comment:
Can `group` be reused where `user` is a nesting?

For example, would `user.effective.group.*` be valid?

Reply:
Yes, but only if it's relevant to the privilege escalation / change / target user.

- [ ] I'll rephrase to explicitly give an example of how group is carried everywhere (including in a self-reuse), and how the self-reuses don't.

Reply:
WDYT about this new presentation? https://ecs_1066.docs-preview.app.elstc.co/guide/en/ecs/master/ecs-user-usage.html#ecs-user-usage-reuse-subtleties